crates/sp1/evm-range-exec/program/src/main.rs

```diff
@@ -62,36 +57,40 @@ pub fn main() {
     // ------------------------------
     // 4. Verify sequential headers (EVM and Celestia)
     // ------------------------------
-
     for window in outputs.windows(2).enumerate() {
         let (i, pair) = window;
         let (prev, curr) = (&pair[0], &pair[1]);
         assert_eq!(
-            curr.prev_header_hash,
-            prev.header_hash,
+            curr.prev_height,
+            prev.new_height,
             "verify sequential EVM headers failed at index {}: expected {:?}, got {:?}",
             i + 1,
-            prev.header_hash,
-            curr.prev_header_hash
+            prev.new_height,
+            curr.prev_height
         );
 
-        // Only check sequentiality if the inclusion height changed.
-        if curr.celestia_header_hash != prev.celestia_header_hash {
-            assert_eq!(
-                curr.prev_celestia_header_hash,
-                prev.celestia_header_hash,
```

Comment:
Previously this meant that if a rollup missed a celestia block height and failed to include something that the assumptions of the proof system would be broken.

The new approach allows support for zero, one or more rollup blocks in the celestia block